plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -1063,12 +1064,23 @@ private class DokkaDescriptorVisitor(
         objectDescriptor(it, dri)
     }
 
-    private fun MemberDescriptor.modifier() = when (modality) {
-        Modality.FINAL -> KotlinModifier.Final
-        Modality.SEALED -> KotlinModifier.Sealed
-        Modality.OPEN -> KotlinModifier.Open
-        Modality.ABSTRACT -> KotlinModifier.Abstract
-        else -> KotlinModifier.Empty
+    private fun MemberDescriptor.modifier(): KotlinModifier {
+        val isInterface = this is ClassDescriptor && this.kind == ClassKind.INTERFACE
+        return if (isInterface) {
+            when (modality) {
+                // modifiers other than "sealed" are redundant for interfaces
+                Modality.SEALED -> KotlinModifier.Sealed
+                else -> KotlinModifier.Empty
+            }
+        } else {
+            when (modality) {
+                Modality.FINAL -> KotlinModifier.Final
+                Modality.SEALED -> KotlinModifier.Sealed
+                Modality.OPEN -> KotlinModifier.Open
+                Modality.ABSTRACT -> KotlinModifier.Abstract
+                else -> KotlinModifier.Empty
+            }
+        }
```

Comment:
This particular bit could certainly be improved, but it's going to be re-written to K2 anyway, and I'd like to minimize merge conflicts with the refactoring I'm doing in a neighbouring branch.